### PR TITLE
Network and listeners review

### LINKED ISSUE #674: node error recovery bug
1. Initialize the app with no network: Node is declared unhealthy
2. Turn network on: Node is still unhealthy, and it should be set to healthy

### LINKED ISSUE #675: nis2 listening
optimization! nis2 listening object, such as the object was created multiple times

### LINKED ISSUE #687: upgrade SDK to v0.15.1

### LINKED ISSUE #692: UI: Network loading indicator
When the user switches node:
- when the endpoint is OK, there is a little time before data is fetched from the network,
- when the endpoint is not OK, Network.ts will throw after about 6 seconds (Listeners could throw earlier, but it is not yet implemented)

During these times I think we should have a simple loading indicator

### PR DESCRIPTION
**Pull request by decentraliser**
**PLEASE TRY TO BREAK IT BEFORE APPROVAL**

- Refactor network and listeners methods, fixes #675:
- => Create `Network` and `Listeners` classes, 
- => Clean-up `App.vue`
- => Add unit tests

- Add generationHash and endpoint checks when setting data related to the network. fixes #674 (prevent from bugs due to setting data after async calls)

- Upgrade to SDK v0.15.1, fixes #687

- Add network loading indicator, fixes #692

### CODE REVIEW

File: src/core/model/Network.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+   private async switchGenerationHash(): Promise<void> {
+      await this.setNetworkMosaics()
+   }
+
+   private async setNetworkMosaics(): Promise<void> {
+      const {store} = this
+      const currentEndpoint = `${this.endpoint}`
+      const firstTx = await this.blockHttp.getBlockTransactions('1', new QueryParams(100)).toPromise()
+      const mosaicDefinitionTx: any[] = firstTx.filter(({type}) => type === TransactionType.MOSAIC_DEFINITION)
+      const mosaicAliasTx: any[] = firstTx.filter(({type}) => type === TransactionType.MOSAIC_ALIAS)
```
Comment (evias) [inline src/core/model/Network.ts]: not sure this is really what we want ; Ultimately these values will be retrievable by REST

File: src/store/account/index.ts [hunk 2]
```diff
@@ -206,5 +202,25 @@ const mutations: MutationTree<StoreAccount> = {
     },
 }
 
+const actions = {
+    SET_GENERATION_HASH({commit, rootState}, {endpoint, generationHash}) {
```
Comment (evias) [inline src/store/account/index.ts]: this is a network action?

File: src/store/account/index.ts [hunk 3]
```diff
@@ -206,5 +202,25 @@ const mutations: MutationTree<StoreAccount> = {
...
+        if (endpoint !== rootState.account.node) return
+        commit('SET_GENERATION_HASH', generationHash)
+    },
+    SET_NETWORK_CURRENCY({commit, rootState}, {endpoint, networkCurrency}) {
```
Comment (evias) [inline src/store/account/index.ts]: this is a network action?

File: src/store/account/index.ts [hunk 4]
```diff
@@ -206,5 +202,25 @@ const mutations: MutationTree<StoreAccount> = {
...
+        if (endpoint !== rootState.account.node) return
+        commit('SET_NETWORK_CURRENCY', networkCurrency)
+    },
+    SET_NETWORK_MOSAICS({commit, rootState}, {endpoint, appMosaics}) {
```
Comment (evias) [inline src/store/account/index.ts]: this is a network action?

File: src/core/model/Network.ts (hunk 1 above)
Comment (decentraliser) [reply src/core/model/Network.ts]: This is what we had till now, and I added a full unit test, for nem.xem and cat.currency + harvest setup.

I agree it is hacky, but it does the job

File: src/store/account/index.ts (hunk 2 above)
Comment (decentraliser) [reply src/store/account/index.ts]: the architecture of the store has to be reviewed, I propose to let it like that for now

**Pull request merged**